sunkit_instruments/rhessi/rhessi.py

```diff
@@ -350,3 +351,53 @@ def _build_energy_bands(label, bands):
     unit = matched.group('UNIT').strip()
 
     return [f'{band} {unit}' for band in bands]
+
+
+def hsi_fits2map(image_datacube):
+    """
+    Extracts single map images from a RHESSI flare image datacube.
+    Currently assumes input to be 4D.
```

Comment:
Is it worth adding a check to ensure that the input is 4D?

Reply:
I have not (yet) encountered 2 or 3D hessi-datacubes tbh, but the [original function](http://www.heliodocs.com/xdoc/xdoc_print.php?file=$SSW/hessi/idl/gen/hsi_fits2map.pro) does handle them... a check and a respective message sure wouldn't hurt. If anyone has a use case and sample files I could even implement it

Reply:
Try studying this codebase and check maybe use the `WCS` object converted from the FITS file to ascertain... 

Some maybe too-much-for-now references (mainly Papers I and II): https://fits.gsfc.nasa.gov/fits_wcs.html

Reply:
We normally do the conversion with some `astropy` magic: https://docs.astropy.org/en/stable/wcs/

Reply:
Think the test concerned is intended for a usually 4D RHESSI flare image datacube for some reason...

@DanRyanIrish If you have the time could you please elaborate on this for some clarification? Are all RHESSI flare image datacubes supposedly 4D by construction or design? I couldn't find any relevant info on the Internet from the official docs at first glance for some reason.

Reply:
Yes they will be by design. The RHESSI image cubes consist of cubes in the shape (n_times, n_energy, image_arr, image_arr) created in the RHESSI GUI.

Reply:
I think incl. the checks below should be good, and the WCS stuff will be handled in the sunpy.map.Map

Reply:
Cool... And I still use raw `astropy` wcs in lieu of `sunpy.map.Map` for my work. Danny never complained about this though 🤣

Reply:
> Cool... And I still use raw `astropy` wcs in lieu of `sunpy.map.Map` for my work. Danny never complained about this though 🤣

In ndcube, which is a lower level set of functionalities than sunpy, it is better to be more hands on with the astropy WCS infrastructure.  But in sunpy or sunkit-instruments, which are higher level and can depend on sunpy, it's perfectly fine to use the sunpy classes (e.g. `sunpy.map.Map`) that handle the WCS parsing for you.